ipatests/test_xmlrpc/test_host_plugin.py

```diff
@@ -337,6 +341,98 @@ def test_add_host_with_null_password(self, host):
         host.track_create()
         host.check_create(result)
 
+    @staticmethod
+    def modify_config_maxhostname(host_tracker, value):
+        try:
+            command = host_tracker.make_command(
+                'config_mod',
+                **dict(
+                    setattr=u'ipamaxhostnamelength={}'.format(value)))
+            command()
+        except errors.EmptyModlist:
+            pass
+
+    @staticmethod
+    def generate_hostname(total_length):
+        if total_length < 9:
+            raise ArithmeticError("Total DNS length in theses tests"
+                                  "must be at least 9")
+        return "aaa.bbb.{}".format((total_length - 8) * "c")
+
+    def test_config_maxhostname_invalid(self):
+        """Change config maxhostname to an invalid value
+        (lower than MAXHOSTNAMELEN). Should fail"""
+        testhost = HostTracker(name=u'testshost')
+        with raises_exact(errors.ValidationError(
+                name='ipamaxhostnamelength',
+                error='must be at least {}'.format(MAXHOSTNAMELEN))):
+            self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN // 2)
+        testhost.ensure_missing()
+
+    def test_raise_hostname_limit_above_maxhostnamelen(self):
+        """Raise hostnamelimit above default. Should pass"""
+        testhost = HostTracker(name=u'testshost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN * 2)
+        testhost.ensure_missing()
+
+    def test_try_hostname_length_above_maxhostnamelimit(self):
+        """Try to create host with hostname length above
+        hostnamelength limit. Should fail"""
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN + 1)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        host_create_command = testhost.make_create_command()
+        with raises_exact(errors.ValidationError(
+                name=u'hostname',
+                error=u'can be at most {} characters'.format(
+                    MAXHOSTNAMELEN))):
+            host_create_command()
+        testhost.ensure_missing()
+
+    def test_try_hostname_length_bellow_maximum(self):
+        """Try to create host with valid hostname. Should pass"""
+        valid_length = MAXHOSTNAMELEN // 2
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(valid_length)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        testhost.delete()
+        host_create_command = testhost.make_create_command()
+        host_create_command()
```

Comment:
Instead of
```
        host_create_command = testhost.make_create_command()
        host_create_command()
```
you can use
```
        testhost.create()
```
This is equivalent but will also update the tracker's internal state to `exists=True`, and make the subsequent call to `ensure_missing` work.